packages/dds/tree/src/tableSchema.ts

```diff
@@ -66,14 +70,49 @@ export namespace TableSchema {
 
 	/**
 	 * Factory for creating new table column schema.
-	 * @privateRemarks
-	 * TODO Add overloads to make propsSchema optional.
 	 * @internal
 	 */
-	// eslint-disable-next-line @typescript-eslint/explicit-function-return-type -- Return type is too complex to be reasonable to specify
+	export function createColumn<const TInputScope extends string | undefined>(
+		inputSchemaFactory: SchemaFactoryAlpha<TInputScope>,
+	): ReturnType<
+		typeof createColumnInternal<
+			TInputScope,
+			FieldSchema<FieldKind.Optional, typeof SchemaFactoryAlpha.null>
+		>
+	>;
+	/**
+	 * Factory for creating new table column schema.
+	 * @internal
+	 */
 	export function createColumn<
 		const TInputScope extends string | undefined,
 		const TPropsSchema extends ImplicitFieldSchema,
+	>(
+		inputSchemaFactory: SchemaFactoryAlpha<TInputScope>,
+		propsSchema: TPropsSchema,
+	): ReturnType<typeof createColumnInternal<TInputScope, TPropsSchema>>;
+	/** `createColumn` implementation */
+	export function createColumn<
+		const TInputScope extends string | undefined,
+		const TPropsSchema extends ImplicitFieldSchema = ImplicitFieldSchema,
+	>(
+		inputSchemaFactory: SchemaFactoryAlpha<TInputScope>,
+		propsSchema?: TPropsSchema,
+	): TreeNodeSchema {
+		// The cast is safe in this case because we know `propsSchema` will only ever be `undefined` when no type was provided for `TPropsSchema`.
+		const props =
+			propsSchema ?? (inputSchemaFactory.optional(inputSchemaFactory.null) as TPropsSchema);
+		return createColumnInternal<TInputScope, TPropsSchema>(inputSchemaFactory, props);
+	}
```

Comment:
I think this would be simpler
```suggestion
	// `createColumn` implementation
	export function createColumn(
		inputSchemaFactory: SchemaFactoryAlpha,
		propsSchema: ImplicitFieldSchema = inputSchemaFactory.optional(inputSchemaFactory.null),
	): TreeNodeSchema {
		return createColumnInternal(inputSchemaFactory, props);
	}
```